framework/aster-frame/src/task/task.rs

```diff
@@ -301,3 +301,29 @@ impl TaskOptions {
         Ok(task)
     }
 }
+
+#[cfg(ktest)]
+mod test {
+    use alloc::boxed::Box;
+    #[ktest]
+    fn create_task() {
+        let task = || {
+            assert_eq!(1, 1);
+        };
+        let task_option = crate::task::TaskOptions::new(task)
+            .data(Box::new(1))
```

Comment:
The `data` field is reserved for passing the pointer of `thread` or `process`. 

In ktest situation, `data` is never used. Ideally, the `data` field should not be set. But it seems that the `build` function of `TaskOption` does not allow this behavior (Personally, I think this should be allowed). 

I suggest set `data` as unit type instead of `Box::new(1)`, which looks a bit magical and requires additional heap allocation.
```rust
let task_option = TaskOptions::new(task).data(())
```

Reply:
Of course, `unreachable!()` is more readable.